pkgs/top-level/python-packages.nix

```diff
@@ -4019,7 +4019,16 @@ in {
 
   jax = callPackage ../development/python-modules/jax { };
 
-  jaxlib = callPackage ../development/python-modules/jaxlib { };
+  jaxlib-bin = callPackage ../development/python-modules/jaxlib/bin.nix { };
+
+  jaxlib-build = callPackage ../development/python-modules/jaxlib {
+    cudaSupport = pkgs.config.cudaSupport or false;
+    cudatoolkit = pkgs.cudatoolkit_11_2;
+    cudnn = pkgs.cudnn_cudatoolkit_11_2;
```

Comment:
Is there a reason to pin this on 11.2 in particular? Why not `pkgs.cudatoolkit`?

Reply:
This is also copy-paste from `tensorflow` build. Officially current TF version supports specifically CUDA 11.2 (see https://www.tensorflow.org/install/gpu). JAX re-uses TF sources and while it doesn't use the same GPU kernels => might not have the same restrictions (https://github.com/google/jax/blob/main/README.md says CUDA 11.x?), I haven't checked this + given JAX is arguably often used together with TF, it's better to avoid pulling in two different CUDA versions in the same program to avoid hard-to-debug issues, so I'd rather stick with the TF-required version?

Reply:
oh good point! let's add a comment here that this should be kept in line with the TF CUDA versions for our future selves. even better if we can find a way to enforce them to be the same

Reply:
Introduced several `tensorflow_compat_*` aliases and switched both `tensorflow` and `jaxlib` to use these, LMK if you think it should be done somehow differently.

Reply:
ok very nice!